LEAF_Request_Portal/sources/System.php

```diff
@@ -289,413 +264,153 @@ public function updateGroup(int $groupID): array
+
+                    // include the backups of employees
+                    $res = $this->db->prepared_query('SELECT * FROM users WHERE userID=:userID AND groupID=:groupID', $vars);
+                    if ($res[0]['active'] == 1) {
+                        $backups = $employee->getBackups($emp['empUID']);
+                        foreach ($backups as $backup) {
+                            $vars = array(':userID' => $backup['userName'],
+                                ':groupID' => $groupID,
+                                ':backupID' => $emp['userName'],);
+
+                            // Add backupID check for updates
+                            $this->db->prepared_query('INSERT INTO users (userID, groupID, backupID)
+                                                        VALUES (:userID, :groupID, :backupID)
+                                                        ON DUPLICATE KEY UPDATE userID=:userID, groupID=:groupID', $vars);
+                        }
+                    }
+                }
+            }
+            $return_value = "groupID: {$groupID} imported";
         }
 
         return $return_value;
     }
 
-    /**
-     * @param int $groupID
-     *
-     * @return array
-     *
-     * Created at: 6/30/2023, 1:25:25 PM (America/New_York)
-     */
-    private function deleteCatPrivs(int $groupID): array
+    public function getServices()
     {
-        $vars = array(':groupID' => $groupID);
-        $sql = 'DELETE
-                FROM `category_privs`
-                WHERE `groupID` = :groupID';
-
-        $return_value = $this->db->pdo_delete_query($sql, $vars);
-
-        return $return_value;
+        return $this->db->prepared_query('SELECT groupID as parentID,
+        							serviceID as groupID,
+        							service as groupTitle,
+        							abbreviatedService as groupAbbreviation
+        							FROM services
+        							ORDER BY groupTitle ASC', array());
     }
 
     /**
-     * @param int $groupID
-     *
-     * @return array
+     * Get the current database version
      *
-     * Created at: 6/30/2023, 1:25:38 PM (America/New_York)
+     * @return string the current database version
      */
-    private function getCatPrivs(int $groupID): array
+    public function getDatabaseVersion()
     {
-        $vars = array(':groupID' => $groupID);
-        $sql = 'SELECT `categoryID`
-                FROM `category_privs`
-                LEFT JOIN `groups` USING (`groupID`)
-                WHERE `category_privs`.`groupID` = :groupID
-                AND `groups`.`groupID` IS NULL';
+        $version = $this->db->prepared_query('SELECT data FROM settings WHERE setting = "dbVersion"', array());
+        if (count($version) > 0 && $version[0]['data'] !== null)
+        {
+            return $version[0]['data'];
+        }
 
-        $return_value = $this->db->pdo_select_query($sql, $vars);
+        return 'unknown';
```

Comment:
Two different return types.  Is that required because of old code?

Reply:
I'm not sure, see my comment on your last comment below.